proposals/0132-sequence-end-ops.md

```diff
@@ -759,211 +204,23 @@ either these axes or other ones. (We would be particularly interested in
-| &nbsp;&nbsp;...with closure      | **S.last(where:)**   | **C.lastIndex(where:)**   | **S.removingLast(where:)**     | **C.removeLast(where:)**     | -            | -                       |
-
-To explain a few entries which might not be immediately obvious: 
-`firstIndex` and `lastIndex` would be `nil` if the collection is empty, 
-and `lastIndex` would be the index before `endIndex`. `prefixIndex` 
-would return the last index of the prefix, and `suffixIndex` would 
-return the first index of the suffix; alternatively, these could be 
-named with `Indices` and return ranges. `first(_:)` and `last(_:)` 
-would return the first and last element equal to the provided value; on 
-a `Set`, they would be roughly equivalent to `NSSet.member(_:)`.
-
-The changes we consider most worthy include:
-
-* Adding corresponding `last` and `suffix` methods for all 
-  `first` and `prefix` methods.
-  
-* Adding corresponding `while:` versions of all appropriate 
-  prefix/suffix APIs.
-
-Ones that could be useful, but can usually be emulated with more work:
-
-* Adding `remove`/`removing`-by-content APIs.
-
-* Adding `prefix/suffixIndex(while:)`.
-
-Ones that are mere conveniences or may not have strong use cases:
-
-* `first/lastIndex` and `prefix/suffixIndex(_:)`.
-
-* `first/last(_:)`.
-
-#### "All" and "Every" as operands
-
-One could imagine adding rows to this table for "all" and "every 
-matching". In addition to creating some useful new API, this would 
-also suggest some interesting renaming for existing APIs:
+### Use just `first` and `last`
 
-* `allIndices` would be a name for `indices`.
+Instead of using `prefix` and `suffix` for multiple elements, we could use 
+`first` and `last` for everything—`first(3)` for the first three elements, 
+etc. This would change fewer names, but the names it would change are 
+probably more frequently used, and it would further overload the `first` 
+and `last` properties with methods, which is confusing and potentially 
+ambiguous. We also think it wouldn't read as clearly. Finally, it would 
+foreclose the use of, for instance, `removeFirst(x)` to remove the first 
+element equal to `x`.
 
-* `removeAll()` is actually an existing name which happens to fit this 
-  pattern.
+### Adding functionality to the standard library
 
-* `every(where:)` would be a name for `filter`. Though some of us 
-  believe `filter` is a strong term of art, we do note that 
-  `every(where:)` does not cause confusion about the sense of its test, 
-  a major complaint about `filter`.
+This renaming exposes some gaps in our standard library functionality. For instance, `removingPrefix(while:)`, `hasPrefix(_:)`, `firstIndex(of:)`, etc. have no end-of-collection equivalents. It's tempting to fill these gaps, but these changes are purely additive and can be made in a future version of Swift.
```

Comment:
Nit: "...can be made in a future version of Swift _if their addition is adequately justified in another proposal_".